### PR TITLE
Assorted typing additions and adjustments

### PR DESCRIPTION
**Pull request by smackesey**
Typing additions/adjustments made while implementing source asset metadata

### CODE REVIEW

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #7006** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/7006" target="_blank">Graphite</a> 
    * **PR #7007** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/7007" target="_blank">Graphite</a>  (this PR)
      * **PR #6900** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/6900" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

File: python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py [hunk 1]
```diff
@@ -46,50 +55,53 @@ def __init__(
...
-    dependedByKeys = non_null_list(GrapheneAssetKey)
     assetMaterializations = graphene.Field(
         non_null_list(GrapheneMaterializationEvent),
         partitions=graphene.List(graphene.String),
         beforeTimestampMillis=graphene.String(),
         limit=graphene.Int(),
     )
-    partitionKeys = non_null_list(graphene.String)
-    partitionDefinition = graphene.String()
+    dependedBy = non_null_list(GrapheneAssetDependency)
```
Comment (sryza) [inline python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py:77]: worth adding a comment that these are alphabetical so people know to add future attrs in the right place

File: python_modules/dagster/dagster/core/definitions/metadata/__init__.py [hunk 2]
```diff
@@ -86,6 +101,12 @@ def normalize_metadata(
     ]
 
 
+import pathlib
```
Comment (sryza) [inline python_modules/dagster/dagster/core/definitions/metadata/__init__.py:104]: Is this on purpose?

File: python_modules/dagster/dagster/core/definitions/output.py [hunk 3]
```diff
@@ -64,35 +62,45 @@ class OutputDefinition:
             partitions from the OutputContext) which should be associated with this OutputDefinition.
     """
 
+    dagster_type: DagsterType
```
Comment (sryza) [inline python_modules/dagster/dagster/core/definitions/output.py:65]: This would make these attributes mutable, which is undesirable.

**Review by sryza — CHANGES_REQUESTED** [review]
Thanks for all the typing additions - makes a lot of code easier to read / reason about.

I won't block it here, and very open to being convinced otherwise, but IMO we shouldn't get into the general habit of alphabetizing attributes / methods everywhere. I don't think the code churn that it will generate is worth the benefit, given that most IDEs make it easy to search for symbols without scrolling through.

File: python_modules/dagster/dagster/core/definitions/output.py (hunk 3 above)
Comment (smackesey) [reply python_modules/dagster/dagster/core/definitions/output.py:65]: reverted

File: python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py (hunk 1 above)
Comment (smackesey) [reply python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py:77]: added

**Review by sryza — APPROVED** [review]

**Pull request merged**